### PR TITLE
Add explanation for compound step methods

### PR DESCRIPTION
**Pull request by huangy22**
Address the issue #2083.

### CODE REVIEW

**Comment by junpenglao** [comment]
Thanks for the PR! However, I think it is better to draft a doc instead of putting the information in the docstring.

**Comment by huangy22** [comment]
Thanks for the reply! I wonder if you could point me to some examples to give me an idea about the format and where should I put this doc? Sorry that I don't have much experience in writing documentations for packages and would really appreciate some help :-)

**Comment by junpenglao** [comment]
You can have a look at this two PRs:
https://github.com/pymc-devs/pymc3/pull/2391
https://github.com/pymc-devs/pymc3/pull/2173

For this doc, I suggest you to put a notebook in https://github.com/pymc-devs/pymc3/tree/master/docs/source/notebooks, and then index it in https://github.com/pymc-devs/pymc3/blob/master/docs/source/examples.rst under Howto

File: pymc3/sampling.py [hunk 1]
```diff
@@ -193,7 +193,21 @@ def sample(draws=500, step=None, init='auto', n_init=200000, start=None, trace=N
            live_plot_kwargs=None, compute_convergence_checks=True, use_mmap=False, **kwargs):
     """Draw samples from the posterior using the given step methods.
 
-    Multiple step methods are supported via compound step methods.
+    Multiple step methods are supported via compound step methods. When compound steps are involved, it takes a list of
```
Comment (junpenglao) [inline pymc3/sampling.py]: Please reverse these changes.

**Comment by junpenglao** [comment]
Look good to me mostly, could you please upgrade your pymc3 version to the newest one and rerun your notebook?

File: pymc3/sampling.py [hunk 2]
```diff
@@ -193,7 +193,7 @@ def sample(draws=500, step=None, init='auto', n_init=200000, start=None, trace=N
            live_plot_kwargs=None, compute_convergence_checks=True, use_mmap=False, **kwargs):
     """Draw samples from the posterior using the given step methods.
 
-    Multiple step methods are supported via compound step methods.
+    Multiple step methods are supported via compound step methods.
```
Comment (twiecki) [inline pymc3/sampling.py]: This isn't required.

**Comment by junpenglao** [comment]
Thanks @huangy22! And congrats on the first contribution ;-)

**Comment by huangy22** [comment]
Thank you for all your help @junpenglao and @twiecki ! :-)

**Pull request merged**